### PR TITLE
Log debug info for the entire build process

### PR DESCRIPTION
**Pull request by FredKSchott**
Log statements, ahoy!

/cc @justinfagnani @azakus

### CODE REVIEW

**Comment by FredKSchott** [comment]
@justinfagnani PTAL

File: src/build/build.ts [hunk 1]
```diff
@@ -137,21 +155,33 @@ export function build(options?: BuildOptions, config?: ProjectConfig): Promise<a
     };
 
     waitForAll([unbundledPhase, bundledPhase])
-      .then(() => analyzer.analyze)
+      .then(() => {
+        return analyzer.analyze;
```
Comment (justinfagnani) [inline src/build/build.ts]: revert?

File: src/build/build.ts [hunk 2]
```diff
@@ -137,21 +155,33 @@ export function build(options?: BuildOptions, config?: ProjectConfig): Promise<a
...
+      })
       .then((depsIndex) => {
+        logger.debug('Analyzing build dependencies...');
```
Comment (justinfagnani) [inline src/build/build.ts]: let put this one a `once` on line 125 with `.pipe(analyzer)`

File: src/build/build.ts [hunk 3]
```diff
@@ -87,31 +96,38 @@ export function build(options?: BuildOptions, config?: ProjectConfig): Promise<a
...
+    logger.debug(`Reading dependencies...`);
     let depsStream =
       vfs.src(dependencies, {cwdbase: true, allowEmpty: true})
         .pipe(depsProject.split)
         .pipe(optimize(optimizeOptions))
         .pipe(depsProject.rejoin)
 
     let allFiles = mergeStream(sourcesStream, depsStream)
-        .pipe(analyzer);
+      .pipe(analyzer);
```
Comment (justinfagnani) [inline src/build/build.ts]: maybe a `.on('end', ...)` for finished analyzing

File: src/commands/build.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export class BuildCommand implements Command {
       return options.env.build(buildOptions, config);
     }
 
-    logger.debug('building via build()...');
+    logger.debug('building via standard build()...');
```
Comment (justinfagnani) [inline src/commands/build.ts]: what's the non-standard build?

File: src/commands/init.ts [hunk 5]
```diff
@@ -76,8 +76,8 @@ export class InitCommand implements Command {
             logger.debug(`Running generator ${generatorName}`);
             env.run(generatorName, {}, () => resolve());
           } else {
-            logger.warn('Template ${options.name} not found');
-            logger.debug('Yeoman generator ${generatorName} not found');
+            logger.warn(`Template ${options.name} not found`);
```
Comment (justinfagnani) [inline src/commands/init.ts]: oops!

**Comment by justinfagnani** [comment]
LGTM with small comments

File: src/build/build.ts (hunk 1 above)
Comment (FredKSchott) [inline src/build/build.ts]: whoops! done.

File: src/build/build.ts (hunk 2 above)
Comment (FredKSchott) [inline src/build/build.ts]: done

File: src/commands/build.ts (hunk 4 above)
Comment (FredKSchott) [inline src/commands/build.ts]: yea I've had trouble with this naming, would love some input. It's the built-in `build` function, as apposed to the `options.env.build` function you can pass via options.

**Pull request merged**